example/reduce/src/reduce.cpp

```diff
@@ -0,0 +1,155 @@
+/**
+ * \file
+ * Copyright 2018 Jonas Schenke, Matthias Werner
+ *
+ * This file is part of alpaka.
+ *
+ * alpaka is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * alpaka is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with alpaka.
+ * If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "alpakaConfig.hpp"
+#include "kernel.hpp"
+#include <alpaka/alpaka.hpp>
+#include <iostream>
+
+// use defines of a specific accelerator
+using Accelerator = CpuSerial;
+
+using DevAcc = Accelerator::DevAcc;
+using DevHost = Accelerator::DevHost;
+using QueueAcc = Accelerator::Stream;
+using Acc = Accelerator::Acc;
+using PltfAcc = Accelerator::PltfAcc;
+using PltfHost = Accelerator::PltfHost;
+using MaxBlockSize = Accelerator::MaxBlockSize;
+
+//-----------------------------------------------------------------------------
+//! Reduces the numbers 1 to n.
+//!
+//! \param devAcc The accelerator object.
+//! \param n The problem size.
+//!
+//! Returns true if the reduction was correct and false otherwise.
+bool reduce(DevAcc devAcc, uint64_t n)
```

Comment:
I would prefer if this was a real reduction function meaning:
* The source buffer is generated outside of this method and passed into it
* The reduction function is passed into this method
* The reuction specific result is check outside of this method
This should all happen within the main function.